pallets/asset/src/lib.rs

```diff
@@ -2209,8 +2212,10 @@ impl<T: Trait> Module<T> {
         ensure!(Self::is_owner(&ticker, did), Error::<T>::Unauthorized);
         // Read the token details
         let mut token = Self::token_details(&ticker);
-        // TODO: either change SecurityToken, or add a new Storage
-        //ensure!(token.is_confidential, Error::<T>::NonConfidentialAssetsCannotSetTotalSupply);
+        ensure!(
+            T::ConfidentialAsset::is_confidential(ticker),
+            Error::<T>::NonConfidentialAssetsCannotSetTotalSupply
```

Comment:
It's not the assets that set the supply but the supply is set for assets. The error name could change accordingly, e.g., `NonConfidentialAssetTotalSupplyCannotBeSet`